tools/rocm-build/docker/ubuntu20/docker_run.sh

```diff
@@ -30,10 +33,14 @@ docker run -ti \
     --mount="type=bind,src=${fake_shadow},dst=/etc/shadow,readonly" \
     --mount="type=bind,src=${fake_group},dst=/etc/group,readonly" \
     -v ${HOME}/.ccache:/.ccache \
+    -v ${HOME}:/home/${CURRENT_USER} \
```

Comment:
No, this is a really really bad thing to do. It means that the user will have their home directory mounted into the build environment, so all their aliases and configuration will be used.

This breaks the idea that the build is the same for everyone. Consider a person who changes their PATH to include ~/bin, and then has their own ~/bin/ln command. The point of using a container to provide the build environment is to make it easy to set up the reproducible build environment.

There are other ways we could set up the build environment, for example we could ship a tarball of a chrooted setup, and tell people to use chroot to enter this subtree. We could use VMs. We could use podman. We are using docker containers for ease of use, but as you can see they are not the only choice. These days this idea seems to be called devcontainers.

We could mount a tmpfs at the $HOME location inside the container. We can also set HOME inside the container to a fixed location. The latter helps with reproducible builds.

Reply:
I see I did it this way because while using the make command the script was failing because it could not find /home/user 
maybe I can just make a directory with the name of the current user instead of mounting it.